src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if ((object.getClass().equals(Rectangle.class))) {
+            Rectangle rectangle = (Rectangle) object;
+            return (rectangle.length == null && this.length == null
+                    || this.length.compareTo(rectangle.length) == 0)
+                    && (rectangle.color == null && this.color == null
+                    || this.color.equals(rectangle.color))
+                    && (rectangle.width == null && this.width == null
+                    || this.width.compareTo(rectangle.width) == 0);
```

Comment:
```suggestion
            return (rectangle.length == null && this.length == null
                    || this.length.compareTo(rectangle.length) == 0)
                    && (rectangle.color == null && this.color == null
                    || this.color.equals(rectangle.color))
                    && (rectangle.width == null && this.width == null
                    || this.width.compareTo(rectangle.width) == 0);
```
There is no need use this keyword if you are calling the field of this class . We should use this keyword in constructors and setters when the method parameter and the field name are the same. But here it's redundant.

Reply:
I have seen your post in Slack about this and removed them, but maybe I forgot to push new version

Reply:
I did remove all thises and pushed (you can check this commit about in the list), dunno how it was in the code.